### ISSUE TITLE
[breaking change] Change the order of arguments in `.add()` function

### ISSUE THREAD

**Issue by taranjeet** [embedchain/embedchain#168]
Right now, add function is
```
app.add("data_type", "url")
```

In the future, we want to move to a point where we only give the URL and it automatically deduces the data type. 
The first step towards this is updating the signature of add function.
```
app.add("url", "data_type")
```

Also the second `data_type` should be an optional or keyworded arg, so that later on it can be omitted. It can be a config also (cc @cachho )

For this, we should update the signature of add function, readme and replit. We can cover code and readme in this PR.

### Deliverables

- [ ] New signature for `.add()`
    `.add(path, type=None)`
    - Args:
        - `path`: takes url, directory path or something else like s3 uri etc in future. We should parse the what the type of `path` is. Eg, determine whether it is a youtube video, url, pdf file etc.
        - `type`: Optional for the user. This holds the type of the data that user is trying to input
- [ ] Remove `.add_local()` function and merge that function inside the `.add()` only after changing the function signature
- [ ] Update documentation and notebooks in the repository based on the above changes

**Comment by cachho** [comment]
Agreed.

**Comment by taranjeet** [comment]
Lets revamp this `add` function as 

- Users can use this embed some data for their bot
- Args:
    - `path`: takes url, directory path or something else like s3 uri etc in future. We should parse the what the type of `path` is. Eg, determine whether it is a youtube video, url, pdf file etc.
    - `type`: Optional for the user. This holds the type of the data that user is trying to input
- Returns id (we will generate the id as part of the ingestion. id will be md5)

**Comment by cachho** [comment]
> * takes url, directory path or something else like s3 uri etc in future. We should parse the what the type of `path` is. Eg, determine whether it is a youtube video, url, pdf file etc.

what's the goal here? Is this trying to unify `add` and `add_local`? Then I feel like `path` isn't describing either. If it's unified I would call it `source`, if we still have local and remote I would stick with `url` and `content` or `source`, especially considering that all remote loaders specifically require a URL.

**Comment by taranjeet** [comment]
We are  aiming for following
* make add smarter with time. This means that in future add will be able to detect the data type on the basis of URL itself. and the first step is to change the signature of the add function.
* add_local will be removed and merged into add. 
Open to suggestion on the parameter front (either of path/content/source/url)

@deshraj @aaishikdutta what do you all think?

**Comment by taranjeet** [comment]
> Why not return a tuple of (added_text, list_of_chunks). That way users can print their own status message, depending on whether they want to see the text or not, want to debug the chunking etc. This has been requested before

A lot of users are not concerned with the chunks. In future if there is interest, we can add a flag in `add` which can return the chunks if users asks for it.

**Comment by cachho** [comment]
> * make add smarter with time. This means that in future add will be able to detect the data type on the basis of URL itself. and the first step is to change the signature of the add function.
I can do that.

> * add_local will be removed and merged into add.
Okay

>   Open to suggestion on the parameter front (either of path/content/source/url)
My vote is with source.

**Comment by cachho** [comment]
> A lot of users are not concerned with the chunks. In future if there is interest, we can add a flag in `add` which can return the chunks if users asks for it.

Fine, I will do that, but what is anyone going to do with the id?
